src/components/Gallery/components/ListItem/index.js

```diff
@@ -17,6 +17,10 @@ const ListItem = ({ item, width, height, selectedIndex, onPress }) => {
   const [baseStyle] = useState(StyleSheet.create({ width, height }))
   const [mutableStyle, setMutableStyle] = useState({})
 
+  const handlePress = useCallback(() => {
+    onPress(item)
+  }, [item])
```

Comment:
I think `onPress` should be in the dependencies array too.
```suggestion
  }, [item, onPress])
```